src/c-dvar-reader.c

```diff
@@ -436,6 +436,21 @@ static int c_dvar_ff(CDVar *var) {
         do {
                 if (var->current->n_type && (var->current->container != 'a' || c_dvar_more(var))) {
                         c = var->current->i_type->element;
+                        if (var->current->container == 'a'
+                                && var->current->i_type->size > 0
+                                && var->current->i_type->basic
+                                && c != 'b') {
+                                /* This is an array of fixed size basic elements other than booleans.
+                                 * We can skip it right now.
+                                 * We just check if all the elements fit in the buffer.
+                                 */
+                                if ((var->current->n_buffer % var->current->i_type->size) != 0)
+                                        return C_DVAR_E_CORRUPT_DATA;
+
+                                var->current->i_buffer += var->current->n_buffer;
+                                var->current->n_buffer = 0;
```

Comment:
Rather than duplicate the test for trailing padding, can we do this:

```
t = var->current->n_buffer % var->current->i_type->size;
var->current->i_buffer += var->current->n_buffer - t;
var->current->n_buffer = t;
```

This will then rely on the `]`-parser to do this check. Thus we avoid hard-coding too much details here.